schema/schema_registry.cc

```diff
@@ -170,9 +170,13 @@ void schema_registry::clear() {
     _entries.clear();
 }
 
-schema_ptr schema_registry_entry::load(frozen_schema fs) {
-    _frozen_schema = std::move(fs);
+schema_ptr schema_registry_entry::load(frozen_base_and_view_schemas fs) {
+    _frozen_schema = std::move(fs.schema);
     auto s = get_schema();
+    if (s->is_view()) {
+        SCYLLA_ASSERT(fs.base_schema);
+        _frozen_base_schema = std::move(fs.base_schema);
+    }
```

Comment:
I decided to use frozen schema because of this:
https://github.com/scylladb/scylladb/blob/98860905d820efa15458458e9c93f148fd81b62f/schema/schema_registry.cc#L324-L335
We seem to transfer the schema between shards here, I believe we shouldn't transfer a `schema_ptr` between shards.

I was thinking about a solution where we keep just `schema_ptr` of the base, but I couldn't come up with a good way to work around the code above.

It probably isn't absolutely necessary to pass both view and base schemas across shards - we still send only the view schema from the registry across nodes in `get_schema_definition`, but looks a fix like that would be much more complicated

Reply:
> We seem to transfer the schema between shards here, I believe we shouldn't transfer a schema_ptr between shards.

We shouldn't, but you could get to frozen schema of the base via view's base schema_ptr's registry entry, like we do for the view via global_schema_ptr::_ptr.

Reply:
Right, wasn't that difficult after all. I replaced the base  `frozen_schema` with `schema_ptr` in the rebase